SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -905,8 +1024,88 @@ private void AccountButton_Click(object sender, RoutedEventArgs e)
 
         private void Slider1Seller_ValueChanged(object sender, RoutedPropertyChangedEventArgs<double> e)
         {
+            TextBoxSliderValue.Text = Slider1Seller.Value.ToString("F2");
             SYLMap?.DrawRadiusOnTempMarker(e.NewValue * 1000.0);
         }
+
+        private void ButtonAddNewMarketPlace_Click(object sender, RoutedEventArgs e)
+        {
+            var timePairTimes = new List<TimePair>();
+            var timePair = new TimePair
+            {
+                StartTime = new Time(TextBoxStartTime.Text),
+                EndTime = new Time(TextBoxEndTime.Text)
+            };
+
+            marketplaceData.AddData(
+                new MarketplaceData(
+                    SYLMap.GetMarkerTempLocation(), TextBoxNewMarketplace.Text.Trim(), null, null, null
+                                    ));
+            marketplaceData.SaveData();
+            ClearUpdateMarketplaceWindow();
+            SYLMap.RemoveAllMarkers();
+            GridMarketplacesAdd.Visibility = Visibility.Collapsed;
+        }
+
+        private void ClearUpdateMarketplaceWindow()
+        {
+            TextBoxNewMarketplace.Text = "";
+            TextBoxStartTime.Text = "";
+            TextBoxEndTime.Text = "";
+        }
+
+        private async void ComboBoxMarketplaceInformatione_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            DeleteAllMarkersExceptTemp();
+            ComboBoxChooseMarketplace.Items.Clear();
+            foreach (var seller in await sellerData.GetAllData())
+            {
+                foreach (var marketplace in await marketplaceData.GetAllData())
```

Comment:
The marketplaceData.GetAllData() function should be called before the for loop, rather than for every cycle, since the data it returns does not depend on the for loop.